capellambse/decl.py

```diff
@@ -59,18 +70,57 @@ def load(file: FileOrPath) -> list[dict[str, t.Any]]:
         An open file-like object, or a path or PathLike pointing to such
         a file. Files are expected to use UTF-8 encoding.
     """
+    with _load(file) as opened_file:
+        return yaml.load(opened_file, Loader=YDMLoader)
+
+
+def load_with_metadata(
+    file: FileOrPath,
+) -> tuple[dict[str, t.Any], list[dict[str, t.Any]]]:
+    """Load a declarative modelling file-stream from a YAML file.
+
+    If the declarative modelling file is missing a metadata section, an
+    empty metadata dict is returned.
+
+    Parameters
+    ----------
+    file
+        An open file-like object, or a path or PathLike pointing to such
+        a file. Files are expected to use UTF-8 encoding.
+
+    Returns
+    -------
+    metadata, instructions
+        A metadata dictionary and a list of instruction dictionaries.
```

Comment:
Badly formatted "Returns" section. Either format metadata and instructions as two separate paragraphs (each with a separate description), or explicitly state that it's a tuple containing these two values.

Also, in order for the correct return type to show up in the generated documentation, the docstring must duplicate the PEP-484 type hints (compare the `function_with_pep484_type_annotations` in the [Napoleon examples](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html#example-numpy)).